everett/manager.py

```diff
@@ -71,6 +101,30 @@ def get_parser(parser):
     return parser
 
 
+def get_key_from_envs(envs, key, namespace=None):
+    """Return the value of a key from the given dict respecting namespaces.
+
+    Data can also be a list of data dicts.
+    """
+    # if it barks like a dict, make it a list
+    # have to use `get` since dicts and lists
+    # both have __getitem__
```

Comment:
The problem is that what we want is an iterator of iterators... maybe we need to give up on this and just always pass a list (or generator).